airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -826,7 +842,11 @@ def on_kill(self) -> None:
             }
             if self.termination_grace_period is not None:
                 kwargs.update(grace_period_seconds=self.termination_grace_period)
-            self.client.delete_namespaced_pod(**kwargs)
+
+            try:
+                self.client.delete_namespaced_pod(**kwargs)
+            except kubernetes.client.exceptions.ApiException:
+                self.log.warning("Pod %s no longer exists", self.pod.metadata.name)
```

Comment:
```suggestion
            except kubernetes.client.exceptions.ApiException:
                self.log.exception("Unable to delete pod %s", self.pod.metadata.name)
```

This can't be the only cause of an ApiException... Maybe this instad?

Reply:
Sure! Thanks!